### PR TITLE
Added tslint configuration for local VSCode

### PR DESCRIPTION
**Pull request by vineetmimrot**
The user needs to do run `npm install` in the root directory and restart VSCode and after that TSLint will start running automatically.

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -10,7 +10,45 @@
...
+			  "import-sources-order": "case-insensitive",
+			  "named-imports-order": "case-insensitive"
+			}
+		],
+		"no-constant-condition": [
+			true,
+			{
+				"checkLoops": false
+			}
+		]
```
Comment (vineetmimrot) [inline tslint.json:50]: I have suppressed many TSLint errors via turning off these rules as these are very basic sanity things and not needed at all. After fixing all the current TSLint errors, we can enable this.

File: tslint.json [hunk 2]
```diff
@@ -10,7 +10,45 @@
...
 	},
-	"defaultSeverity": "warning"
+	"defaultSeverity": "warning",
+	"extends": ["tslint-microsoft-contrib/recommended", "tslint:recommended"]
```
Comment (vineetmimrot) [inline tslint.json:53]: `tslint-microsoft-contrib/recommended` show errors corresponding to ci build scan i.e. sdl required.

`tslint:recommended` show general TSLint errors.

File: package.json [hunk 3]
```diff
@@ -90,18 +90,20 @@
...
-        "@types/js-yaml": "^3.12.1",
         "ajv": "^6.9.1",
         "assert": "1.4.1",
         "mocha": "5.2.0",
-        "tslint": "5.8.0",
         "ts-node": "7.0.1",
+        "tslint": "5.8.0",
+        "tslint-microsoft-contrib": "^6.2.0",
         "typescript": "3.3.1",
+        "typescript-tslint-plugin": "^0.5.5",
```
Comment (bishal-pdMSFT) [inline package.json:106]: for my understanding - what is difference between tslint and typescript-tslint-plugin?

Comment (bishal-pdMSFT) [reply package.json:106]: and where is this plugin used?

File: tslint.json (hunk 1 above)
Comment (bishal-pdMSFT) [reply tslint.json:50]: @vineetmimrot how will we track fixing these warning?

File: tslint.json (hunk 2 above)
Comment (bishal-pdMSFT) [inline tslint.json:53]: if we specify "tslint-microsoft-contrib/recommended" in extends here, 'do we also need to specify rulesDirectory in line no 2? what is the different between the two?

File: tslint.json [hunk 4]
```diff
@@ -10,7 +10,45 @@
...
+		"no-string-based-set-immediate": true,
+		"no-string-based-set-interval": true,
+		"no-string-based-set-timeout": true,
+		"no-eval": true,
+		"quotemark": false,
+		"no-relative-imports": false,
+		"max-line-length": [
+			true,
+			{
+			  "limit": 200,
```
Comment (bishal-pdMSFT) [inline tslint.json:32]: I would prefer keeping a smaller max length

File: package.json (hunk 3 above)
Comment (vineetmimrot) [reply package.json:106]: It is VSCode plugin, which runs the TSLint in the IDE.

File: tslint.json (hunk 2 above)
Comment (vineetmimrot) [reply tslint.json:53]: Line no 2, specifies the rules directory, but it doesn't say which rules to use from the directory.
Extends specifies which rule config files to use and tool will search into rules directory.

File: tslint.json (hunk 4 above)
Comment (vineetmimrot) [reply tslint.json:32]: by default it is 120, I made it 200. 
Will change to 135

**Pull request merged**